src/analyze/analyze-chid.c

```diff
@@ -243,10 +308,19 @@ int verb_chid(int argc, char *argv[], void *userdata) {
         if (r < 0)
                 return r;
 
+        const char *edid_path = strv_find_prefix(strv_skip(argv, 1), "/sys/class/drm/");
```

Comment:
there are multiple ways to access a sysfs device. via the class symlink is just one. hence please sd-device apis, i.e. sd_device_new_from_path() + validate subsystem.